pkg/pod/entrypoint.go

```diff
@@ -182,27 +182,26 @@ func UpdateReady(kubeclient kubernetes.Interface, pod corev1.Pod) error {
 
 // StopSidecars updates sidecar containers in the Pod to a nop image, which
 // exits successfully immediately.
-func StopSidecars(nopImage string, kubeclient kubernetes.Interface, pod corev1.Pod) error {
+func StopSidecars(nopImage string, kubeclient kubernetes.Interface, pod corev1.Pod, containers *[]corev1.Container) (error) {
+	updated := false
 	newPod, err := kubeclient.CoreV1().Pods(pod.Namespace).Get(pod.Name, metav1.GetOptions{})
 	if err != nil {
 		return fmt.Errorf("error getting Pod %q when stopping sidecars: %w", pod.Name, err)
 	}
-
-	updated := false
 	if newPod.Status.Phase == corev1.PodRunning {
 		for _, s := range newPod.Status.ContainerStatuses {
-			// Stop any running container that isn't a step.
-			// An injected sidecar container might not have the
-			// "sidecar-" prefix, so we can't just look for that
-			// prefix.
-			if !IsContainerStep(s.Name) && s.State.Running != nil {
-				for j, c := range newPod.Spec.Containers {
-					if c.Name == s.Name && c.Image != nopImage {
-						updated = true
-						newPod.Spec.Containers[j].Image = nopImage
+			for _, containerItem := range *containers {
+				if s.Name == containerItem.Name && s.State.Running != nil {
+					for j, c := range newPod.Spec.Containers {
+						if c.Name == s.Name && c.Image != nopImage {
+							updated = true
+							fmt.Println(fmt.Sprintf("Attempting to terminate Sidecar container %s since it was not marked for exemption from ForceTermination in pod %s.", s.Name, newPod.Name))
```

Comment:
This should use a logger instead, probably derived from a provided context. This formats it so that it can be more easily indexed/searched, instead of just printing directly to stdout.

Reply:
Totally. I'll check what logger is being used in general and use it

Reply:
I'm using the logger now, but I'm not comfortable initialising empty context logger just for this :( Please suggest if there's a better way.